source/gfoidl.DataCompression/Compression/SwingingDoorCompression.cs

```diff
@@ -367,6 +433,66 @@ public override bool MoveNext()
                         return true;
                 }
             }
+            //---------------------------------------------------------------------
+            /// <summary>
+            /// Returns an array of the compressed <see cref="DataPoint" />s.
+            /// </summary>
+            /// <returns>An array of the compressed <see cref="DataPoint" />s.</returns>
+            public override DataPoint[] ToArray()
+            {
+                TList source      = _source;
+                int snapShotIndex = 0;
+
+                if ((uint)snapShotIndex >= (uint)source.Count)
+                    return Array.Empty<DataPoint>();
+
+                DataPoint snapShot = source[snapShotIndex];
+
+                if (source.Count < 2)
+                    return new[] { snapShot };
+
+                var arrayBuilder = new ArrayBuilder<DataPoint>(true);
+                arrayBuilder.Add(snapShot);
+                this.OpenNewDoor(0);
+
+                int incomingIndex = 1;
+                for (; incomingIndex < source.Count; ++incomingIndex)
+                {
+                    ref var archive = ref this.IsPointToArchive(incomingIndex);
```

Comment:
In `IsPointToArchive` and in `CloseTheDoor` the _DataPoints_ for `lastArchived` and `incoming` are fetched from the array, preceding a (manual) range check. Maybe it's better to store these two _DataPoints_ on the stack, then reuse them.